### DISCUSSION TITLE
Why choose MulanPubL-2.0

### DISCUSSION THREAD

**Discussion by botouzi** [oceanbase/oceanbase#70] [Classroom] [answered]
增加了老外程序员的认知成本，如此高调的一个开源项目，用一个不知名的开源许可，难道只希望在国内发展？
自主可不是啥都使用自己的东西，那是闭门造车。被罚后这么不自信吗？

**Comment by yangzhifeng** [comment]
1. 木兰许可证并非“不知名”的开源许可，只是“你”不知道。例如，“木兰宽松许可”是opensource.org认可的开源许可证，见 https://opensource.org/licenses/MulanPSL-2.0
2. 如果读不懂中文，You can read the english version of the license at https://license.coscl.org.cn/MulanPubL-2.0/
3. 请充分理解此许可证，认可欢迎参与，不认可别参与。

**Comment by xumengpanda** [comment]
It seems MulanPSL v2 allows cloud provider to host the DB as a service. 
Could you confirm if that's the case? 

Thanks

**Comment by MizuhaHimuraki** [comment]
> It seems MulanPSL v2 allows cloud provider to host the DB as a service.
> Could you confirm if that's the case?

Thanks for your feedback, OceanBase is under the **MulanPubL-2.0**, not **MulanPSL v2**.

```
“对应源代码” 是指生成、安装和(对于可执行作品)运行目标代码所需的所有源文件和与之关联的接口定义文件，以及控制这些活动的脚本，但不包括编译环境、编译工具、云服务平台（如果有）。

“分发” 是指通过任何媒介向他人提供“贡献”或“衍生作品”的行为，以及利用“贡献”或“衍生作品”通过网络远程给用户提供服务的行为，例如：通过利用“贡献”或“衍生作品”搭建的云服务平台提供在线服务的行为。

Corresponding Source Code means all the source code needed to generate, install, and (for an executable work) run the object code including the interface definition files associated with source files for the work, and scripts to control those activities, excluding of compilation environment and compilation tools, cloud services platform (if any).
Distribute (or Distribution) means the act of making the Contribution or Derivative Work available to others through any medium, and using the Contribution or Derivative Work to provide online services to users, such as the act of providing online services through a cloud service platform built using Contributions or Derivative Works.
```
Please refers to [LICENSE](https://github.com/oceanbase/oceanbase/blob/master/LICENSE) file in root path or https://license.coscl.org.cn/MulanPubL-2.0/

**Comment by xumengpanda** [comment]
> > It seems MulanPSL v2 allows cloud provider to host the DB as a service.
> > Could you confirm if that's the case?
> 
> Thanks for your feedback, OceanBase is under the **MulanPubL-2.0**, not **MulanPSL v2**.
> 
> ```
> “对应源代码” 是指生成、安装和(对于可执行作品)运行目标代码所需的所有源文件和与之关联的接口定义文件，以及控制这些活动的脚本，但不包括编译环境、编译工具、云服务平台（如果有）。
> 
> “分发” 是指通过任何媒介向他人提供“贡献”或“衍生作品”的行为，以及利用“贡献”或“衍生作品”通过网络远程给用户提供服务的行为，例如：通过利用“贡献”或“衍生作品”搭建的云服务平台提供在线服务的行为。
> 
> Corresponding Source Code means all the source code needed to generate, install, and (for an executable work) run the object code including the interface definition files associated with source files for the work, and scripts to control those activities, excluding of compilation environment and compilation tools, cloud services platform (if any).
> Distribute (or Distribution) means the act of making the Contribution or Derivative Work available to others through any medium, and using the Contribution or Derivative Work to provide online services to users, such as the act of providing online services through a cloud service platform built using Contributions or Derivative Works.
> ```
> 
> Please refers to [LICENSE](https://github.com/oceanbase/oceanbase/blob/master/LICENSE) file in root path or https://license.coscl.org.cn/MulanPubL-2.0/

Thanks for the explanation! 

According to the description, it seems that any software built on top of MulanPubL-2.0 will need to be open-sourced. 

Is the understanding correct?

> 
> 4.   分发限制
> 
> 
> （2）如果您“分发”您接收到的“贡献”，您必须使用“本许可证”提供该“贡献”的源代码副本；如果您 “分发”您的“衍生作品”，您必须：
> 
> （i）随“衍生作品”提供使用“本许可证”“分发”的您的“衍生作品”的“对应源代码”。如果您通过下载链接提供前述“对应源代码”，则您应将下载链接地址置于“衍生作品”或其随附文档中的明显位置，有效期自该“衍生作品”“分发”之日起不少于三年，并确保接收者可以获得“对应源代码”；或者，
> 
> （ii）随“衍生作品”向接收者提供一个书面要约，表明您愿意提供根据“本许可证”“分发”的您“衍生作品”的“对应源代码”。该书面要约应置于“衍生作品”中的明显位置，并确保接收者根据书面要约可获取“对应源代码”的时间从您接到该请求之日起不得超过三个月，且有效期自该“衍生作品”“分发”之日起不少于三年。

**Comment by MizuhaHimuraki** [comment]
> it seems that any software built on top of MulanPubL-2.0 will need to be open-sourced.
> 
> Is the understanding correct?

Of course, unless your software is an undisclosed system.

BTW, you can also refer to some projects that use the AGPLv3 license (such as MongoDB, Grafana, etc.), because MulanPubL-2.0 is compatible with the AGPLv3.

> 如果您将“贡献”与采用GNU AFFERO GENERAL PUBLIC LICENSE Version 3（以下简称“AGPLv3”）或其后续版本的作品结合形成新的“衍生作品”，且根据“AGPLv3”或其后续版本的要求您有义务将新形成的“衍生作品”以“AGPLv3”或其后续版本进行许可的，您可以根据“AGPLv3”或其后续版本进行许可
>
> If you combine Contribution or your Derivative Work with a work licensed under the GNU AFFERO GENERAL PUBLIC LICENSE Version 3 (hereinafter referred to as “AGPLv3”) or its subsequent versions, and according to the AGPLv3 or its subsequent versions, you have an obligation to make the combined work to be licensed under the corresponding license, you can license such combined work under the license, provided that when you Distribute the combined work, you also provide a copy of this License to the recipients, and retain copyright, trademarks, patents, and disclaimer statements in the Contribution. No Contributor will grant additional rights to the recipients of the combined work for your license under AGPLv3 or its subsequent versions.

**Comment by longdafeng** [comment, marked as answer]
If you take a deep look at MulanPubL-2.0, you will find it is better than GPL . 
What's more MulanPubL-2.0 meet our requirement. 

In our mind,  basing GPL,  MulanPubL-2.0 add some limitation in the case Software as a service, this is for cloud vendor.

**Comment by luchsh** [comment]
> 增加了老外程序员的认知成本，如此高调的一个开源项目，用一个不知名的开源许可，难道只希望在国内发展？
> 自主可不是啥都使用自己的东西，那是闭门造车。被罚后这么不自信吗？

为啥要在乎“增加了老外程序员的认知成本”？欧美开源这帮人用英文难道不是增加了东亚人的认知成本？
OB，做好自己最重要！
